src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,162 @@
+import classNames from 'classnames';
+import React, {
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import * as api from '../../api/todos';
+import { Todo } from '../../types/Todo';
+import { TodoLoader } from '../TodoLoader';
+import { TodoUpdateContext, TodosContext } from '../../context/TodosContext';
+import { Errors } from '../../types/Errors';
+
+interface Props {
+  todoItem: Todo,
+}
+
+export const TodoItem: React.FC<Props> = ({ todoItem }) => {
+  const { id, title, completed } = todoItem;
+  const {
+    errorMessage,
+    setErrorMessage,
+    setTodos,
+    setLoadingIds,
+  } = useContext(TodosContext);
+  const {
+    deleteTodo,
+    toggleTodo,
+    // editTodo,
+  } = useContext(TodoUpdateContext);
+
+  const [loading, setLoading] = useState(false);
+  const [editTitle, setEditTitle] = useState(title);
+  const [isEditing, setIsEditing] = useState(false);
+
+  const titleField = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    titleField.current?.focus();
+  }, [isEditing, errorMessage]);
+
+  const handleDelete = async () => {
+    try {
+      await deleteTodo(id);
+    } catch (error) {
+      setErrorMessage(Errors.Delete);
+    }
+  };
+
+  const handleCheckbox = async () => {
+    try {
+      setLoading(true);
+
+      const updatedTodo = { ...todoItem, completed: !completed };
+
+      await toggleTodo(updatedTodo);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const saveEditing = () => {
+    if (!editTitle.trim()) {
```

Comment:
create the variable, that includes the trimmed value of Title.  You are using `.trim()` a few times.